nscplugin/src/main/scala-3/scala/scalanative/nscplugin/AdaptLazyVals.scala

```diff
@@ -0,0 +1,132 @@
+package scala.scalanative.nscplugin
+
+import dotty.tools._
+import dotc._
+import dotc.transform.{LazyVals, MoveStatics}
+import dotc.ast.tpd._
+import plugins._
+import core.Flags._
+import core.Contexts._
+import core.Names._
+import core.Symbols._
+import core.StdNames._
+import scala.annotation.{threadUnsafe => tu}
+
+// This phase is responsible for rewriting calls to scala.runtime.LazyVals with
+// its scala native specific counter-part. This is needed, becouse LazyVals are
+// using JVM unsafe API and static class constructors which are not supported
+// in Scala Native.
+object AdaptLazyVals extends PluginPhase {
+  val phaseName = "scalanative-adaptLazyVals"
+
+  override val runsAfter = Set(LazyVals.name, MoveStatics.name)
+  override val runsBefore = Set(GenNIR.phaseName)
+
+  def defn(using Context) = LazyValsDefns.get
+  def defnNir(using Context) = NirDefinitions.defnNir
+
+  private def isLazyFieldOffset(name: Name) =
+    name.startsWith(nme.LAZY_FIELD_OFFSET.toString)
+
+  // Map of field symbols for LazyVals offsets and literals
+  // with the name of referenced bitmap fields within given TypeDef
+  private val bitmapFieldNames = collection.mutable.Map.empty[Symbol, Literal]
+
+  override def prepareForUnit(tree: Tree)(using Context): Context = {
+    bitmapFieldNames.clear()
+    super.prepareForUnit(tree)
+  }
+
+  // Collect informations about offset fields
+  override def prepareForTypeDef(td: TypeDef)(using Context): Context = {
+    val sym = td.symbol
+    val hasLazyFields = sym.denot.info.fields
+      .exists(f => isLazyFieldOffset(f.name))
+
+    if (hasLazyFields) {
+      val template @ Template(_, _, _, _) = td.rhs
+      bitmapFieldNames ++= template.body.collect {
+        case vd: ValDef if isLazyFieldOffset(vd.name) =>
+          val Apply(_, List(cls: Literal, fieldname: Literal)) = vd.rhs
+          vd.symbol -> fieldname
+      }.toMap
+    }
+
+    ctx
+  }
+
+  // Replace all usages of all unsuportted LazyVals methods with their
+  // Scala Native specific implementation (taking Ptr intead of object + offset)
```

Comment:
```suggestion
  // Replace all usages of all unsupported LazyVals methods with their
  // Scala Native specific implementation (taking Ptr instead of object + offset)
```